plugin/contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -632,12 +632,18 @@ func (k *deleter) deleteOne(pod *Pod) error {
 }
 
 // Create creates a scheduler plugin and all supporting background functions.
-func (k *KubernetesScheduler) NewPluginConfig(terminate <-chan struct{}, mux *http.ServeMux) *PluginConfig {
+func (k *KubernetesScheduler) NewPluginConfig(terminate <-chan struct{}, mux *http.ServeMux,
+	makePodsWatcher func() *cache.ListWatch) *PluginConfig {
```

Comment:
I would recommend changing the 3rd argument to `*cache.ListWatch` and making it required, not optional. Then add a `NewDefaultPluginClonfig` (or more informative name)  with only 2 args that delegates to `NewPluginClonfig` with the result of `createAllPodsLW(k.client)`.